challenges/revertable/src/index.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import useRevertable from './useRevertable';
+
+function App({ realValue }) {
+  const { pendingValue, value, cancel, submit, status, typing } = useRevertable(
```

Comment:
Per the original challenge stated, we don't have the `status` nor the `pendingValue` states, why do we need them in this context?

Reply:
The statement said:
* When onChange is invoked: value is changed but not realValue
* When onSubmit is invoked: realOnChange is invoked and thus realValue changed.
* When onCancel is invoked: value be reverted to realValue

realValue won't change unless we click submit button (trigger onSubmit), this behavior is pretty much the same like pending, the naming of `realValue` confused me little bit.
So in this context, I choose `status` to keep track of what current status is. If it's `pending`, don't reflect to realValue.

Reply:
But yes, maybe it complicated the original statement though, depends on what use case.

Reply:
I think the original challenge implies that `realValue` is a state from a parent component above (or simply another `useState`), so `submit` is not actually changing the internal state of `Revertable` or `useRevertable`, but calling the handler in the parent component to update the `realValue` state.

IMHO it's a little confusing to have 3 different values in the same hook (`realValue`, `value`, `pendingValue`). Also `status` is another learning obstacle that need to be documented and used correctly.

Reply:
Sure, make sense to me

Reply:
Hi, do you still want to make changes for this?

Reply:
I rewrite the implementation, in this case I feel like we don't have to create a custom hook, render prop would be easier in my opinion.